habitat/plan.sh

```diff
@@ -0,0 +1,116 @@
+  export OPENSSL_INCLUDE_DIR=$(pkg_path_for openssl)/include
+  export SSL_CERT_FILE=$(pkg_path_for cacerts)/ssl/cert.pem
+
+  build_line "Setting link for /usr/bin/env to 'coreutils'"
+  [[ ! -f /usr/bin/env ]] && ln -s $(pkg_path_for coreutils)/bin/env /usr/bin/env
+
+  return 0
+}
+
+do_build() {
+  export CPPFLAGS="${CPPFLAGS} ${CFLAGS}"
+
+  local _bundler_dir=$(pkg_path_for bundler)
+  local _libxml2_dir=$(pkg_path_for libxml2)
+  local _libxslt_dir=$(pkg_path_for libxslt)
+  local _zlib_dir=$(pkg_path_for zlib)
+
+  export GEM_HOME=${pkg_prefix}
+  export GEM_PATH=${_bundler_dir}:${GEM_HOME}
+
+  export NOKOGIRI_CONFIG="--use-system-libraries --with-zlib-dir=${_zlib_dir} --with-xslt-dir=${_libxslt_dir} --with-xml2-include=${_libxml2_dir}/include/libxml2 --with-xml2-lib=${_libxml2_dir}/lib"
+  bundle config --local build.nokogiri '${NOKOGIRI_CONFIG}'
+
+  bundle config --local silence_root_warning 1
+
+  bundle install --without dep_selector --no-deployment --jobs 2 --retry 5 --path $pkg_prefix
+
+  bundle exec 'rake install:local'
+}
+
+do_install() {
+
+  mkdir -p $pkg_prefix/bin
+
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin chef-dk
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin chef
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin ohai
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin foodcritic
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin test-kitchen
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin berkshelf
+  bundle exec appbundler $HAB_CACHE_SRC_PATH/$pkg_dirname $pkg_prefix/bin inspec
+
+  if [[ `readlink /usr/bin/env` = "$(pkg_path_for coreutils)/bin/env" ]]; then
+    build_line "Removing the symlink we created for '/usr/bin/env'"
+    rm /usr/bin/env
+  fi
+
+  wrap_ruby_bin "$pkg_prefix/bin/chef"
+  wrap_ruby_bin "$pkg_prefix/bin/chef-client"
+  wrap_ruby_bin "$pkg_prefix/bin/chef-apply"
+  wrap_ruby_bin "$pkg_prefix/bin/chef-shell"
+  wrap_ruby_bin "$pkg_prefix/bin/chef-solo"
+  wrap_ruby_bin "$pkg_prefix/bin/ohai"
+  wrap_ruby_bin "$pkg_prefix/bin/knife"
+  wrap_ruby_bin "$pkg_prefix/bin/kitchen"
+  wrap_ruby_bin "$pkg_prefix/bin/berks"
+  wrap_ruby_bin "$pkg_prefix/bin/foodcritic"
+  wrap_ruby_bin "$pkg_prefix/bin/inspec"
+}
+
+# Stubs
+do_strip() {
+  return 0
+}
+
+# Copied from https://github.com/habitat-sh/core-plans/blob/f84832de42b300a64f1b38c54d659c4f6d303c95/bundler/plan.sh#L32
+wrap_ruby_bin() {
+  local bin="$1"
+  build_line "Adding wrapper $bin to ${bin}.real"
+  mv -v "$bin" "${bin}.real"
+  cat <<EOF > "$bin"
+#!$(pkg_path_for busybox-static)/bin/sh
+set -e
+if test -n "$DEBUG"; then set -x; fi
+export GEM_HOME="$pkg_prefix/ruby/2.4.0/"
+export GEM_PATH="$(hab pkg path core/ruby)/lib/ruby/gems/2.4.0:$(hab pkg path core/bundler):$pkg_prefix/ruby/2.4.0/:$GEM_HOME"
+export SSL_CERT_FILE=$(hab pkg path core/cacerts)/ssl/cert.pem 
```

Comment:
Anyone know if there's some better way of doing this?

Reply:
Not that I've found.

Reply:
openssl has now been rebuilt with that path baked in, so this _should_ be unnecessary AIUI.